src/routes/balances/balances.controller.spec.ts

```diff
@@ -830,13 +830,13 @@ describe('Balances Controller (Unit)', () => {
 
       await request(app.getHttpServer())
         .get(`/v1/chains/${chainId}/safes/${safeAddress}/balances/usd`)
-        .expect(500)
+        .expect(503)
         .expect({
-          statusCode: 500,
-          message: 'Internal server error',
+          code: 503,
+          message: 'Service unavailable',
         });
 
-      expect(networkService.get.mock.calls.length).toBe(4);
+      expect(networkService.get.mock.calls.length).toBe(3);
```

Comment:
These will be addressed by [adjusting the `HttpErrorFactory`](https://github.com/safe-global/safe-client-gateway/pull/2100/files#diff-6393ab1bfb621198efe5705d010d4bcd4991e56ff2eb0b2f062942d70346f20aR18).

Reply:
Sounds good to me, and I'm pretty sure you are aware, but just something to have in mind: I'd say we need to be cautious there and avoid exposing the details of a ZodError when the validation failure comes from an error while processing a Datasource response (e.g.: a Transaction Service response), to avoid leaking data through the public route controller 👍🏻

Reply:
Totally agree with you!